test/integration/consul-container/libs/cluster/container.go

```diff
@@ -111,6 +116,14 @@ func NewConsulContainer(ctx context.Context, config Config, cluster *Cluster, po
 		return nil, fmt.Errorf("error chowning data directory %s: %w", tmpDirData, err)
 	}
 
+	if config.ExternalDataDir != "" {
+		// copy consul persistent state from an external dir
+		err := copy.Copy(config.ExternalDataDir, tmpDirData)
```

Comment:
This seems fine since we're making the temporary directory on the fly, but I'm curious if it makes sense to just use the given `config.ExternalDataDir` below as the `dataDir` value and skip creating a temporary directory and having to copy the contents to it here? If we didn't need to copy the data, we wouldn't need the new dependency and this code could be skipped.

Just a thought.

Reply:
Thanks for the comments. I was thinking using the ExternalDataDir as well. however, I think it's better copying the data because

1. The container test framework will create another files in `tmpDirData` such as the tls files, so generating these file might not be possible in `ExternalDataDir`.
2. `ExternalDataDir` could be reused for the test where it comes from, so keeping a copy might be easier for separating the state between these tests.